bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,239 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.FloureonDevice;
+import com.github.mob41.blapi.dev.hysen.AdvancedStatusInfo;
+import com.github.mob41.blapi.dev.hysen.BaseStatusInfo;
+import com.github.mob41.blapi.dev.hysen.SensorControl;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.hysen.SetTimeCommand;
+
+/**
+ * The {@link FloureonThermostatHandler} is responsible for handling thermostats labeled as Floureon Thermostat.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@NonNullByDefault
+public class FloureonThermostatHandler extends BroadlinkThermostatHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FloureonThermostatHandler.class);
+    @Nullable
+    private FloureonDevice floureonDevice;
+
+    /**
+     * Creates a new instance of this class for the {@link FloureonThermostatHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+        try {
+            blDevice = new FloureonDevice(host, new Mac(mac));
+            this.floureonDevice = (FloureonDevice) blDevice;
+        } catch (IOException e) {
+            logger.error("Could not find broadlinkthermostat device at Host {} with MAC {} ", host, mac, e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command class: {}", command.getClass());
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
+            return;
+        }
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case SETPOINT:
+                handleSetpointCommand(channelUID, command);
+                break;
+            case POWER:
+                handlePowerCommand(channelUID, command);
+                break;
+            case MODE:
+                handleModeCommand(channelUID, command);
+                break;
+            case SENSOR:
+                handleSensorCommand(channelUID, command);
+                break;
+            case REMOTE_LOCK:
+                handleRemoteLockCommand(channelUID, command);
+                break;
+            case TIME:
+                handleSetTimeCommand(channelUID, command);
+                break;
+            default:
+                logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handlePowerCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            try {
+                floureonDevice.setPower(command == OnOffType.ON);
+            } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type? Please check all.

Reply:
The Exception is thrown by the broadlink-java-api library and unfortunately that one does not throw a specific exception.

Reply:
Ok, I overlooked that.